pkgs/development/tools/verible/default.nix

```
@@ -0,0 +1,82 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, buildBazelPackage
+, bazel_4
+, flex
+, bison
+, python3
+}:
+
+buildBazelPackage rec {
+  pname = "verible";
+  version = "v0.0-2172-g238b6df6";
+
+  # These environment variables are read in bazel/build-version.py to create
+  # a build string. Otherwise it would attempt to extract it from .git/.
+  GIT_DATE = "2022-08-08";
+  GIT_VERSION = "${version}";
+
+  src = fetchFromGitHub {
+    owner = "chipsalliance";
+    repo = "verible";
+    rev = "${version}";
+    sha256 = "sha256-iOJhdbipuqqBiYGgk95d1c8bEK6Z16l16GuzYCQRc2g=";
+  };
+
+  # Patch WORKSPACE file to not include windows-related dependencies,
+  # as they are removed by bazel, breaking the fixed output derivation
+  patches = [ ./remove-unused-deps.patch ];    # TODO: fix upstream
+
+  bazelFlags = [ "--//bazel:use_local_flex_bison" ];
+
+  fetchAttrs = {
+    # Fixed output derivation hash after bazel fetch
+    sha256 = if stdenv.isDarwin then
+        "sha256-JjibpdSkz3eiO/PCRx0D7h5u3Q8R44InRyqW9mJSMHo="
+       else
+        "sha256-XoLdlEeoDJlyWlnXZADHOKu06zKHgHJfgey8UhOt+LM=";
+  };
+
+  nativeBuildInputs = [
+    flex       # We use local flex and bison as WORKSPACE sources fail
+    bison      # .. to compile with newer glibc
+    python3
+  ];
+
+  postPatch = ''
+    patchShebangs .
```

```suggestion
    patchShebangs WORKSPACE
```
if only one file need patching, we should only patch that one. That is a lot faster.

---

oh, the shebangs need to be patched all over, there are a bunch of bash and python files in here.

I thought you meant the `patches = []` section.

---

Do we have a common directory where those files are? We should just avoid patching literally every file.

---

They are not in a common directory, there are in the order of 45 files in 15 directories involved. Is there a way to narrow the patchShebang pattern with a recursive match of `**/*.py`, `**/*.sh` ?